pkg/logging/logger.go

```diff
@@ -46,11 +46,14 @@ func NewLogger(logLevel, logFormat, debugName string) log.Logger {
 		logger = log.NewJSONLogger(log.NewSyncWriter(os.Stderr))
 	}
 
+	// Sort the logger chain to avoid expensive log.Valuer evaluation for disallowed level.
+	// Ref: https://github.com/go-kit/log/issues/14#issuecomment-945038252
+	logger = log.With(logger, "ts", log.DefaultTimestampUTC, "caller", log.Caller(5))
```

Comment:
Why is it `log.Caller(5)` now instead of `log.Caller(3)` (which is the default caller)?

Reply:
Because the NewFilter wraps next(the logger) and implements level filtering.

> Note that you cannot use log.DefaultCaller in this case, because the level filter separates the log.With that sets up the caller Valuer from the calls to log.With inside the level.Info and level.Debug calls. So there are two additional stack frames for log.Caller to skip before getting back to the call-site of the Log methods.

Reply:
I see, that's smart and totally makes sense :bow: